cli/src/main/java/org/jboss/galleon/cli/HelpSupport.java

```diff
@@ -168,6 +170,77 @@ private static String pad(int max, String k) {
         return lst;
     }
 
+    public static String getCommandTree(CommandRegistry<? extends Command, ? extends CommandInvocation> registry,
+                                        String command) {
+        CommandLineParser<? extends Command> cmdParser;
+        try {
+            cmdParser = registry.getCommand(command, null).getParser();
+        } catch (CommandNotFoundException ex) {
+            return "";
+        }
+
+        StringBuilder sb = new StringBuilder();
+        ProcessedCommand<? extends Command> processedCommand = cmdParser.getProcessedCommand();
+        String processedDescription = processedCommand.description();
+
+        if (processedDescription.contains(".")) {
+            processedDescription = processedDescription.substring(0, processedDescription.indexOf("."));
+        }
+        sb.append(processedCommand.name()).append(" : ").append(processedDescription);
+        sb.append(getCommandOptions(processedCommand, 0));
+
+        if (!cmdParser.getAllChildParsers().isEmpty()) {
+            for (CommandLineParser<? extends Command> childParser : cmdParser.getAllChildParsers()) {
+                ProcessedCommand<? extends Command> childProcessedCommand = childParser.getProcessedCommand();
+                processedDescription = childProcessedCommand.description();
+                if (processedDescription.contains(".")) {
+                    processedDescription = processedDescription.substring(0, processedDescription.indexOf("."));
+                }
+                sb.append(Config.getLineSeparator()).append("    ");
+                sb.append(childProcessedCommand.name()).append(" : ").append(processedDescription);
+                sb.append(getCommandOptions(childProcessedCommand, 4));
+            }
+        }
+
+        return sb.toString();
+    }
+
+    public static String getHelpFooter() {
```

Comment:
When updating HelpSupport.getHelpCommandHelp, you shouldn't need this method.

Reply:
I think we can retire `getHelpCommandHelp()` since the help is built differently with this new implementation. Wdyt?